src/Identity/Swauth/Service.php

```diff
@@ -42,14 +39,34 @@ public static function factory(ClientInterface $client): self
      */
     public function authenticate(array $values): array
     {
-        $response = $this->execute(self::TOKEN_DEFINITION, array_intersect_key($values, self::TOKEN_DEFINITION['params']));
 
-        $token = $this->model(Token::class, $response);
+        if (!empty($values['cachedToken'])) {
+            $token = $this->generateTokenFromCache($values['cachedToken']);
```

Comment:
Consider to factor out these if/else bodies to separate (private) methods, so this function can be reduced to
```php
if (!empty($values['cachedToken'])) {
    return $this->reuseToken($values['cachedToken']);
}
return $this->createNewToken();
```